CONTRIBUTORS.md

```diff
@@ -9,17 +9,62 @@ to make via an [issue](https://github.com/keylime/keylime/issues) or email to th
    outlining the fix or feature.
 2. Fork the keylime repository to your own github account and clone it locally.
 3. Hack on your changes.
-4. Update the README.md or documentation with details of changes to any
-   interface, this includes new environment variables, exposed ports, useful
-   file locations, CLI parameters and configuration values.
-5. Add and commit your changes with some descriptive text on the nature of the
-   change / feature in your commit message. Also reference the issue raised at
-   [1] as follows: `Fixes #45`. See [here](https://help.github.com/articles/closing-issues-using-keywords/)
-   for more message types
+4. Update the README.md or documentation in [keylime-docs](https://github.com/keylime/keylime-docs)
+   with details of changes to any interface, this includes new environment
+   variables, exposed ports, useful file locations, CLI parameters and
+   new or changed configuration values.
+5. Correctly format your commit message see [Commit Messages](#Commit Message Guidelines)
+   below.
 6. Ensure that CI passes, if it fails, fix the failures.
 7. Every pull request requires a review from the [core keylime team](https://github.com/orgs/keylime/teams/core)
    before merging.
 
+## Commit Message Guidelines
+
+We follow the commit formatting recommendations found [here](https://chris.beams.io/posts/git-commit/)
+
+Well formed commit messages not only help reviewers understand the nature of
+the Pull Request, but also assists the release process where commit messages
+are used to generate release notes.
+
+A good example of a commit message would be as follows:
+
+```
+Summarize changes in around 50 characters or less
+
+More detailed explanatory text, if necessary. Wrap it to about 72
+characters or so. In some contexts, the first line is treated as the
+subject of the commit and the rest of the text as the body. The
+blank line separating the summary from the body is critical (unless
+you omit the body entirely); various tools like `log`, `shortlog`
+and `rebase` can get confused if you run the two together.
+
+Explain the problem that this commit is solving. Focus on why you
+are making this change as opposed to how (the code explains that).
+Are there side effects or other unintuitive consequences of this
+change? Here's the place to explain them.
+
+Further paragraphs come after blank lines.
+
+ - Bullet points are okay, too
+
+ - Typically a hyphen or asterisk is used for the bullet, preceded
+   by a single space, with blank lines in between, but conventions
+   vary here
+
+If you use an issue tracker, put references to them at the bottom,
+like this:
+
+Resolves: #123
+See also: #456, #789
+```
+
+Note the `Resolves #123` tag, this references the issue raised and allows us to
+ensure issues are associated and closed when a pull request is merged.
+
+See [here](https://help.github.com/articles/closing-issues-using-keywords/) for
```

Comment:
Might be good to use a more descriptive text than just "here" for the link.   

Could we change this to something like:
See GitHub's [Closing issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/) article for more message types.

Reply:
check the next commit, added a better desc.